arwen/host/execution.go

```diff
@@ -137,6 +137,13 @@ func (host *vmHost) checkGasForGetCode(input *vmcommon.ContractCallInput, meteri
 	return nil
 }
 
+// doRunSmartContractDelete delete a contract directly
+func (host *vmHost) doRunSmartContractDelete(input *vmcommon.ContractCallInput) *vmcommon.VMOutput {
+	vmOutput := host.outputContext.GetVMOutput()
```

Comment:
add checkPermission here. Do the same in the upgrade function as well. Let's check permission before doing the actual action, not after.